frame/staking/src/pallet/mod.rs

```diff
@@ -982,7 +984,9 @@ pub mod pallet {
 			}
 
 			Self::do_remove_nominator(stash);
-			Self::do_add_validator(stash, prefs);
+			Self::do_add_validator(stash, prefs.clone());
+			Self::deposit_event(Event::<T>::CommissionChanged(ledger.stash, prefs.commission));
```

Comment:
wouldn't this also emit an even if the commission itself didnt change?

I was expecting something like reading the existing commission, then comparing it to the new one. Providing both values in the event also is helpful so that the front-end can show if the value has increased or decreased compared to before.

Reply:
The scenario is that front-end applications may want to notify users if they detect that a validator has increased their commission, since that may affect how a nominator may view and select that validator.

Reply:
Two thoughts here:
1. A validator should not call this function if it were not for a commission change, therefore it's unlikely they would call it twice with the same commission.
2. There are no other events deposited in this call, so it could be useful for it to fire in case apps want to sync validator prefs using this event.
 
Would it not make sense for front end apps to check whether the commission has changed, and act accordingly if it does? Rather than the node assuming front ends only want updated values (if they have any to begin with)

Reply:
Is there a possibility in the future more ValidationPrefs will be introduced? In which case we should change the event name to `ValidationPrefsUpdated`

Reply:
> A validator should not call this function if it were not for a commission change, therefore it's unlikely they would call it twice with the same commission.

Validator Prefs also has `blocked`, so your PR does not cover all the scenarios.

Reply:
@shawntabrizi are you happy with the updated `ValidatorPrefsUpdated` event to work with commission changes?